src/callbacks/liquidity/Ramses/RamsesV1DTL.sol

```diff
@@ -0,0 +1,195 @@
+    ) BaseDirectToLiquidity(auctionHouse_) {
+        if (pairFactory_ == address(0)) {
+            revert Callback_Params_InvalidAddress();
+        }
+        if (router_ == address(0)) {
+            revert Callback_Params_InvalidAddress();
+        }
+        pairFactory = IRamsesV1Factory(pairFactory_);
+        router = IRamsesV1Router(router_);
+    }
+
+    // ========== CALLBACK FUNCTIONS ========== //
+
+    /// @inheritdoc BaseDirectToLiquidity
+    /// @dev        This function implements the following:
+    ///             - Validates the parameters
+    ///
+    ///             This function reverts if:
+    ///             - The pool for the token combination already exists
+    function __onCreate(
+        uint96 lotId_,
+        address,
+        address baseToken_,
+        address quoteToken_,
+        uint256,
+        bool,
+        bytes calldata callbackData_
+    ) internal virtual override {
+        // Decode the callback data
+        RamsesV1OnCreateParams memory params = abi.decode(callbackData_, (RamsesV1OnCreateParams));
+
+        // Check that the pool does not exist
+        if (pairFactory.getPair(baseToken_, quoteToken_, params.stable) != address(0)) {
+            revert Callback_Params_PoolExists();
+        }
+
+        // Record whether the pool should be stable or volatile
+        lotIdToStable[lotId_] = params.stable;
+    }
+
+    /// @inheritdoc BaseDirectToLiquidity
+    /// @dev        This function implements the following:
+    ///             - Validates the parameters
+    ///             - Creates the pool if necessary
+    ///             - Deposits the tokens into the pool
+    function _mintAndDeposit(
+        uint96 lotId_,
+        address quoteToken_,
+        uint256 quoteTokenAmount_,
+        address baseToken_,
+        uint256 baseTokenAmount_,
+        bytes memory callbackData_
+    ) internal virtual override {
+        // Decode the callback data
+        RamsesV1OnSettleParams memory params = abi.decode(callbackData_, (RamsesV1OnSettleParams));
+
+        // Create and initialize the pool if necessary
+        // Token orientation is irrelevant
+        bool stable = lotIdToStable[lotId_];
+        address pairAddress = pairFactory.getPair(baseToken_, quoteToken_, stable);
+        if (pairAddress == address(0)) {
+            pairAddress = pairFactory.createPair(baseToken_, quoteToken_, stable);
+        }
+
+        // Calculate the minimum amount out for each token
+        uint256 quoteTokenAmountMin = _getAmountWithSlippage(quoteTokenAmount_, params.maxSlippage);
+        uint256 baseTokenAmountMin = _getAmountWithSlippage(baseTokenAmount_, params.maxSlippage);
+
+        // Approve the router to spend the tokens
+        ERC20(quoteToken_).approve(address(router), quoteTokenAmount_);
+        ERC20(baseToken_).approve(address(router), baseTokenAmount_);
+
+        // Deposit into the pool
+        // Token orientation is irrelevant
+        router.addLiquidity(
```

Comment:
we can depend on this reverting if the token is liquid and the pool is initialized at a price different from the auction. at that point, the auction would fail to settle and a refund could be issued with abort.

Reply:
Added a note on this